File: src/main/java/com/moira/itda/global/entity/CommonCode.java
package com.moira.itda.global.entity;

import com.moira.itda.domain.common.dto.request.CommonCodeAddRequest;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.ZonedDateTime;

@Builder
@Getter
@Setter
public class CommonCode {
    private String key;
    private ZonedDateTime createdAt;
    private ZonedDateTime updatedAt;

    public static CommonCode fromCommonCodeAddRequest(CommonCodeAddRequest request) {
        return CommonCode.builder()
                .key(request.codeKey())
                .createdAt(ZonedDateTime.now())
                .updatedAt(ZonedDateTime.now())
                .build();
    }
}

File: src/main/kotlin/com/moira/itda/global/entity/CommonCode.kt
package com.moira.itda.global.entity

import com.moira.itda.domain.admin.code.dto.request.AdminCodeAddRequest
import java.time.ZonedDateTime

data class CommonCode(
    val codeKey: String,
    val createdAt: ZonedDateTime,
    val updatedAt: ZonedDateTime
) {
    companion object {
        fun fromCommonCodeAddRequest(request: AdminCodeAddRequest): CommonCode {
            return CommonCode(
                codeKey = request.key,
                createdAt = ZonedDateTime.now(),
                updatedAt = ZonedDateTime.now()
            )
        }
    }
}
